### PR TITLE
feat: add piggybacker

### PR DESCRIPTION
**Pull request by souradeep-das**

### CODE REVIEW

File: ife-challenger/src/piggybacker.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                  // check if IFE is in first phase
+                  const exitId = await rootChain.getInFlightExitId(
+                    { txBytes: ife.details.txbytes },
+                  );
+                  const exitData = await rootChain.getInFlightExitData({ exitIds: [exitId] });
+                  const blockNumber = await web3.eth.getBlockNumber();
+                  const { timestamp } = await web3.eth.getBlock(blockNumber);
+                  const periodTime = exitPeriod / 2;
+
+                  if ((timestamp - exitData[0].exitStartTimestamp) / periodTime < 1) {
```
Comment (souradeep-das) [inline ife-challenger/src/piggybacker.js:40]: should we use BN here?

Comment (kevsul) [reply ife-challenger/src/piggybacker.js:40]: I think there are a few million years left before we run out of space for a timestamp in an integer. 
But if you're taking a particularly long term view, go for it 😄

File: ife-challenger/src/piggybacker.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+          const ife = ifes[i];
+
+          // Check if any of the ifes are quasar ifes
+          const { inputs } = transaction.decodeTxBytes(ife.details.txbytes);
+          for (let j = 0; j < inputs.length; ++j) {
+            const utxoPos = transaction.encodeUtxoPos(inputs[j]);
+            const isQIFE = await quasar.isQuasarIfe(utxoPos);
+            if (isQIFE) {
+              for (let k = 0; k < ife.details.available_outputs.length; ++k) {
+                // check if the first output is not piggybacked
```
Comment (kevsul) [inline ife-challenger/src/piggybacker.js:29]: Can we add a comment in here clearly explaining that the Quasar only accepts the _first_ output in a claim tx.

File: ife-challenger/src/piggybacker.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+                    await rootChain.piggybackInFlightExitOnOutput({
+                      inFlightTx: ife.details.txbytes,
+                      outputIndex: 0,
+                      txOptions: {
+                        privateKey: quasarOwner.privateKey,
+                        from: quasarOwner.address,
+                      },
+                    });
+                  } else {
+                    console.log(`Skipped Piggyback IFE ${utxoPos.toString()}`);
```
Comment (kevsul) [inline ife-challenger/src/piggybacker.js:52]: Is this potentially an error if the output has not been piggybacked and it's in the 2nd exit phase?
Maybe change this to a warning with a more descriptive message

**Review by kevsul — APPROVED** [review]
Looks good, couple of comments

**Comment by souradeep-das** [comment]
kept the timestamp the same then 😝

**Pull request merged**